### PR TITLE
Outsource tokenise and spacing

### PR DESCRIPTION
**Pull request by lorenzwalthert**

### CODE REVIEW

File: R/nested.R [hunk 1]
```diff
@@ -37,6 +36,18 @@ compute_parse_data_nested <- function(text) {
   pd_nested
 }
 
+#' Obtain token from text
```
Comment (krlmlr) [inline R/nested.R]: token table?

File: R/nested.R [hunk 2]
```diff
@@ -37,6 +36,18 @@ compute_parse_data_nested <- function(text) {
...
+#'
+#' [utils::getParseData()] is used to obtain a flat parse table from `text`.
+#' @param text A character vector.
+#' @return A flat parse table
+tokenize <- function(text) {
+  parsed <- parse(text = text, keep.source = TRUE)
+  parse_data <- tbl_df(utils::getParseData(parsed, includeText = TRUE))
```
Comment (krlmlr) [inline R/nested.R]: I think `as_tibble()` is preferred these days. Do we still need `includeText = TRUE`?

File: R/nested.R [hunk 3]
```diff
@@ -37,6 +36,18 @@ compute_parse_data_nested <- function(text) {
   pd_nested
 }
 
+#' Obtain token from text
+#'
+#' [utils::getParseData()] is used to obtain a flat parse table from `text`.
+#' @param text A character vector.
+#' @return A flat parse table
+tokenize <- function(text) {
```
Comment (krlmlr) [inline R/nested.R]: Do we have another instance of `getParseData()` that can be replaced with a `tokenize()` call?

Comment (lorenzwalthert) [inline R/nested.R]: oh yeah for the flat calculations. Replaced that.

Comment (lorenzwalthert) [inline R/nested.R]: Should I move it to R/utils.R then or should it stay where it is?

Comment (krlmlr) [inline R/nested.R]: It's fine where it is now.

File: R/nested.R (hunk 2 above)
Comment (lorenzwalthert) [inline R/nested.R]: ok

File: R/nested.R (hunk 1 above)
Comment (lorenzwalthert) [inline R/nested.R]: ok.

File: R/nested.R (hunk 2 above)
Comment (lorenzwalthert) [inline R/nested.R]: technically no, but `includeText = TRUE` is sometimes convenient so you can see everything inside a token without going through the whole nested structure. But I think we could also use `create_node_from_nested_root` for that so I can set the argument to `NA`.

**Pull request merged**